src/parties/Party/party.js

```diff
@@ -258,6 +289,34 @@ class Party extends AMaaSModel {
       throw new Error('Not a valid email')
     }
   }
+  
+ /**
+ * Upsert an Phone Number
+ * @param {string} type - Type of Phone Number (e.g. 'Work', 'Support')
+ * @param {PhoneNumbers} phoneNumber - new phoneNumber. Note that the new phoneNumber cannot be primary if a primary phoneNumber already exists. Use this.phoneNumbers setter to replace primary phoneNumbers (??)
+ */
+  upsertPhoneNumber(type, phoneNumber) {
+    const phoneNumbers = Object.assign({}, this.phoneNumbers)
+    if (phoneNumber.phoneNumberPrimary) {
+      for (let ref in phoneNumbers) {
+        if (phoneNumbers.hasOwnProperty(ref)) {
+          phoneNumbers[ref].phoneNumberPrimary = false
+        }
+      }
+    }
+    phoneNumbers[type] = phoneNumber
+    this.phoneNumbers = phoneNumbers
+  }
+
+  // Check if input is a valid phoneNumber string
+  _checkPhoneNumber(phoneNumber) {
```

Comment:
I don't think this is being used anymore right? Anyway I don't think we need validation on the phone number format at this stage.